### PR TITLE
[lgwebos] Added IRCode channel to simulate infrared remote control

### PR DESCRIPTION
**Pull request by bbrodt**
[lgwebos] Add IRCode Channel

This PR adds a new channel ("IRCode") to the LG WebOS binding, which simulates *some* of the button presses on the LG infrared remote control. Inspiration for this enhancement came from [the discussion here.](https://community.openhab.org/t/lgwebos-binding-sendbutton-question/85872/8
)

Here is a list of IR codes that I have found to work. Note that this list has been compiled mostly through trial and error, and I may have missed some of the buttons. I have only been able to test this on an LGUJ657A TV so your mileage may vary.

| Code String | Description                                              |
|-------------|----------------------------------------------------------|
| LEFT        | Left button in cursor control group                      |
| RIGHT       | Right button in cursor control group                     |
| UP          | Up button in cursor control group                        |
| DOWN        | Down button in cursor control group                      |
| ENTER       | "OK" button in the center of the cursor control group    |
| BACK        | "BACK" button                                            |
| EXIT        | "EXIT" button                                            |
| 0-9         | Number buttons                                           |
| HOME        | "HOME" button                                            |
| RED         | "RED"  button                                            |
| GREEN       | "GREEN" button                                           |
| YELLOW      | "YELLOW" button                                          |
| BLUE        | "BLUE" button                                            |
| PLAY        | "PLAY" button                                            |
| PAUSE       | "PAUSE" button                                           |
| STOP        | "STOP" button                                            |

A sample HABPanel remote control widget can be found [in this github repository.](https://github.com/bbrodt/openhab2-misc)

### CODE REVIEW

File: bundles/org.openhab.binding.lgwebos/README.md [hunk 1]
```diff
@@ -53,6 +53,33 @@ Please note that at least one channel must be bound to an item before the bindin
 | mediaPlayer     | Player    | Media control player                                                                                                                                                                                                    | W          |
 | mediaStop       | Switch    | Media control stop                                                                                                                                                                                                      | W          |
 | appLauncher     | String    | Application ID of currently running application. This also allows to start applications on the TV by sending a specific Application ID to this channel.                                                                 | RW         |
+| irCode          | String    | Simulates pressing of a button on the TV's infrared remote control. See below for a list of control strings.                                                                                                            | W          |
```
Comment (sprehn) [inline bundles/org.openhab.binding.lgwebos/README.md]: irCode stands for Infra Red Code I suppose. Should we not name it Remote Control Button or sth similar. I am not sure if the remote actually uses infra red or another form of connection nowadays. The value which go over this channel certainly won't be the codes used by the remote internally.

**Comment by wepanx** [comment]
Would this allow us to power on a TV (2014er webos device w/o WOL functionality)?

I guess I know the answer, but anyway... 😉

**Comment by bbrodt** [comment]
> Would this allow us to power on a TV (2014er webos device w/o WOL functionality)?
> 
> I guess I know the answer, but anyway... 😉

Nope, sorry! You still need WOL.

File: bundles/org.openhab.binding.lgwebos/README.md (hunk 1 above)
Comment (t-8ch) [reply bundles/org.openhab.binding.lgwebos/README.md]: All these commands are sent to device as mouse buttons. Why not call it "button" instead?

Comment (t-8ch) [reply bundles/org.openhab.binding.lgwebos/README.md]: Well, it's already fixed. Sorry for the noise.
The fact that the PR has 80 commits and 50.000 changed lines makes it hard to review.

**Comment by davidgraeff** [comment]
This PR is flawed. It contains a lot of changes outside of the lgwebos directory.

**Comment by bbrodt** [comment]
Abandoned after request for name change/rebase

**Pull request closed without merging**